README.md

```diff
@@ -1,71 +1,157 @@
 
-### Dependencies
+## Use Cases & Expected Behavior
 
-1. [go](https://golang.org/doc/install)
+### Deploying your product using Knative Service
 
-### Setup environment
+KCD supports progressive deployment if you deploy your services using Knative Service. You should specify the name of a rollout policy (see below) using annotations like so:
 
-Put the following in a `./bashrc` or `./bashprofile`
+```yaml
+apiVersion: serving.knative.dev/v1alpha1
+kind: Service
+metadata:
+  ...
+  annotations:
+    delivery.knative.dev/policy: <your_policy>
+...
+```
+
+Note that you must first create a valid policy with the correct name, otherwise the progressive rollout will not initiate because it will not be able to find the correct policy resource with the given name. The newest Revision created for this Service will follow the policy you specified, slowly increasing its traffic percentage over time. All the older Revisions will have their routing percentages adjusted as appropriate.
+
+### Deploying your product using Knative Route and Configuration
+
+If you are not using Knative Service to deploy your service, KCD also supports Route and Configurations (if you use Knative Service, then Route and Configuration will be automatically created by Knative). You should annotate your Configuration resource with the correct policy name (as shown above). Note that your Route and Configuration MUST have the same namespace and name, otherwise KCD will not be able to associate your Route and Configuration correctly, and progressive rollout will fail.
+
+### Deployment multiple Revisions simultaneously
+
+KCD can handle more than 2 active Revisions for a service at the same time. It will always satisfy the traffic demands of the latest Revision first, followed by the second latest Revision, third latest Revision, and so on. When 100% of traffic has been assigned to a set of Revisions, any older Revisions that are not included in this pool will be eliminated as a traffic target. An in-depth example demonstrating the routing behavior when multiple Revisions are simultaneously "in flight" can be found [here](https://docs.google.com/document/d/1UhMW-NTml1ywPD_PttteCQgIoglNdFK27jrU1NirHuQ/edit?usp=sharing)
+
+## Rollout Policies
+
+### Built-in Policies
+
+KCD comes with some predefined, example policies for you to play with. One example is the following yaml manifest of a policy:
+
+```yaml
+apiVersion: delivery.knative.dev/v1alpha1
+kind: Policy
+metadata:
+  namespace: default
+  name: example-policy1
+spec:
+  mode: time
+  defaultThreshold: 60
```

Comment:
nit: in this example the defaultThreshold is never used.

Reply:
fixed by rewriting this example